nitro-protocol/contracts/InterestBearingApp.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity 0.8.17;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/IForceMoveApp.sol';
+import './libraries/NitroUtils.sol';
+import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+
+// InterestBearingApp is a ForceMoveApp that allows a lender to earn interest
+// on a deposit. It functions as a ConsensusApp with the following additional rule:
+//   - the lender can unilaterally transition from state n to state n+1,
+//     forcing calculated interest into the lender's Outcome allocation
+contract InterestBearingApp is IForceMoveApp {
+    struct Funds {
+        address[] asset;
+        uint256[] amount;
+    }
+
+    struct InterestAppData {
+        // the per-block interest rate, expressed as the denominator of a fraction
+        // eg, 1% per block would be expressed by interestPerBlockDivisor = 100,
+        //     2% per block would be expressed by interestPerBlockDivisor = 50,
+        //     0.1% per block would be expressed by interestPerBlockDivisor = 1000,
+        // etc.
+        uint256 interestPerBlockDivisor;
+        // the block number of the latest principal adjustment
+        uint256 blocknumber;
+        // the current principal. Decreases as the borrower earns via the channel.
+        Funds principal;
+        // the total interest collected so far. Strictly increasing.
+        // The value of the lender's allocation can never be less than this:
+        // ie, when the lender's collectedInterest grows to be equal to their
+        // allocation, the channel is effectively spent and can be concluded.
+        Funds collectedInterest;
+    }
+
+    enum AllocationIndicies {
+        borrower, // intends to earn service fees up to a limit of the financier's deposit
+        lender // makes initial deposit and earns interest
+    }
+
+    function requireStateSupported(
+        FixedPart calldata,
+        RecoveredVariablePart[] calldata proof,
+        RecoveredVariablePart calldata candidate
+    ) external view override {
+        if (proof.length == 0) {
+            // unanimous consensus check
+            require(
+                NitroUtils.getClaimedSignersNum(candidate.signedBy) == 2,
```

Comment:
Do we enforce that there are only two participants in the channel?  Or need a stronger consensus check like [ConensusApp](https://github.com/statechannels/go-nitro/blob/7490468c0a5fed0cc64c6fa1c68c30760566315a/nitro-protocol/contracts/libraries/signature-logic/Consensus.sol#L19)?

Reply:
I guess my concern is that it would be pretty easy for a `lender` to include another participant they control to the channel. Since the App only needs 2 participants to support a channel, they could unilateral support any channel they want.

Reply:
The core protocol provides protection against that class of attack. Challenges against a particular cID are vetted so that the provided FixedPart has the correct participant list. Signatures attached to a challenge state do not "get counted" unless the recovered signer is in the participant list.

Convincing myself of all this did lead me to discover [a very closely related attack](https://github.com/statechannels/go-nitro/issues/1176), where the lender (or any participant 0) just stuffs a bunch of their OWN signature into the process.